src/test/java/com/commercetools/paymenttoorderprocessor/helper/CartAndMessageCreateHelper.java

```diff
@@ -0,0 +1,110 @@
+import static com.commercetools.paymenttoorderprocessor.fixtures.PaymentFixtures.EUR_20;
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class CartAndMessageCreateHelper {
+
+    @Autowired
+    BlockingSphereClient testClient;
+
+    @Autowired
+    private ApplicationContext context;
+
+    @Autowired
+    private MessageFilter messageProcessor;
+
+    /**
+     * Create a cart and attach the {@code payment} to it, set successful payment change transaction for the
+     * {@code payment} and wait after related message is created.
+     *
+     * At the end - fetch the message and return {@link CartAndMessage} object with these values.
+     *
+     * The operation is blocking.
+     *
+     * @param payment payment to process
+     * @return {@link CartAndMessage} with created cart and successful payment state change message.
+     */
+    public  CartAndMessage createCartAndMessage(Payment payment) {
+        //Preconditions: to create message in commercetools platform
+        final PagedQueryResult<TaxCategory> result = testClient.executeBlocking(TaxCategoryQuery.of().byName("standard"));
+        final List<TaxCategory> results = result.getResults();
+        assertThat(results).isNotEmpty();
+        final CustomLineItemDraft customLineItemDraft = CustomLineItemDraft.of(LocalizedString.ofEnglish("messageProcesserIntegrationTestCustomLineItem"), "Slug", EUR_20, results.get(0), 1L);
+        final Address address = Address.of(CountryCode.DE);
+        final Cart cart = CartFixtures.createCart(testClient, CartDraft.of(EUR)
+                .withCustomLineItems(Collections.singletonList(customLineItemDraft))
+                .withShippingAddress(address));
+        final Cart cartWithPayment = testClient.executeBlocking(CartUpdateCommand.of(cart, AddPayment.of(payment)));
+
+        final TransactionDraft transactionDraft = TransactionDraftBuilder.of(TransactionType.AUTHORIZATION, EUR_20).build();
+        final AddTransaction addTransaction = AddTransaction.of(transactionDraft);
+
+        final Payment paymentWithTransaction = testClient.executeBlocking(PaymentUpdateCommand.of(payment, addTransaction));
+        assertThat(paymentWithTransaction.getTransactions().get(0).getState()).isEqualTo(TransactionState.PENDING);
+
+        final Transaction transaction = paymentWithTransaction.getTransactions().get(0);
+        final ChangeTransactionState changeTransactionState = ChangeTransactionState.of(TransactionState.SUCCESS, transaction.getId());
+        final Payment paymentWithTransactionStateChange = testClient.executeBlocking(PaymentUpdateCommand.of(paymentWithTransaction, changeTransactionState));
+
+        //final array so lambda can use it
+        final PaymentTransactionStateChangedMessage[] message = new PaymentTransactionStateChangedMessage[1];
+
+        //Give Platform time to create messages
+        try {
+            Thread.sleep(10000L);
```

Comment:
probably would be better to wait in a tests in a loop for message instead of this here but we can change it if we have flaky tests results